elastic/beats
# [filebeat] add 8.x kibana logs ingest pipeline

### Summary
Closes elastic/kibana#31216

Added a separate ingest pipeline to handle kibana 8.x log formats. The pipeline triggers when the `ecs.version` field exists and otherwise falls back to the 7.x pipeline.

The pipeline reproduces what is done for the elasticsearch 8.x logs by [unwrapping all the fields under `json.` to the root](https://github.com/elastic/beats/blob/main/filebeat/module/elasticsearch/deprecation/ingest/pipeline-json-8.yml) of the document since the format is ecs compatible. Due to potential mapping explosion with the request/response headers, these fields are copied from `http.request.headers` to `kibana.log.meta.req.headers` where we have control over the field definition. Here we define them as [flattened](https://www.elastic.co/guide/en/elasticsearch/reference/current/flattened.html)

### Testing
- Add logging configuration to kibana.yml
```
logging.root.level: debug

logging:
  appenders:
    file:
      type: file
      fileName: ./kibana.log
      layout:
        type: json
  root:
    appenders: [file]
```
- Enable kibana filebeat module
```
filebeat.modules:
  - module: kibana
    log:
      enabled: true
      var.paths:
        - ../../kibana/kibana.log
```
- Logs are successfully ingested. Note that filebeat can output failure to parse some `Right` logs which is handled in elastic/beats#31576
